packages/smooth_app/lib/pages/prices/infinite_scroll_controller.dart

```diff
@@ -0,0 +1,95 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+/// A generic controller for handling infinite scrolling in lists.
+/// [T] is the type of items being displayed.
+/// [P] is the type of parameters used for pagination.
+class InfiniteScrollController<T, P> {
+  InfiniteScrollController({
+    required this.fetchItems,
+    required Iterable<T> initialItems,
+    this.initialPage = 1,
```

Comment:
Correct me if I'm wrong, but you never set this field to something.
Therefore it's always 1.
Please simplify the code accordingly, making this field a `const` if relevant. That's a good practice anyway to be explicit about the starting point (0 or 1).